gpytorch/likelihoods/multitask_gaussian_likelihood.py

```diff
@@ -51,12 +86,19 @@ def forward(self, input):
             added.
         """
         mean, covar = input.representation()
-        eye_lv = DiagLazyVariable(torch.ones(covar.size(-1) // self.n_tasks, device=self.log_noise.device))
-        task_var_lv = DiagLazyVariable(self.log_task_noises.exp())
+        eye_lv = DiagLazyVariable(
+            torch.ones(covar.size(-1) // self.n_tasks, device=self.log_noise.device)
+        )
+        if hasattr(self, "log_task_noises"):
+            task_var_lv = DiagLazyVariable(self.log_task_noises.exp())
+        else:
+            task_var_lv = RootLazyVariable(self.task_noise_covar_factor)
         diag_kron_lv = KroneckerProductLazyVariable(task_var_lv, eye_lv)
```

Comment:
I think the way we are doing it is the right way. In the `rank = 0` case, we fit `D_t` that is not the identity matrix (or a scaled identity). Since we are just generalizing `D_t` when `rank > 0` we would preserve the order of operations.